include/ada/url_base.h

```diff
@@ -172,101 +77,13 @@ namespace ada {
      */
     ada_really_inline size_t parse_port(std::string_view view, bool check_trailing_content = false) noexcept;
 
-    /**
-     * @private
-     * Take the scheme from another URL. The scheme string is moved from the provided url.
-     */
-    inline void copy_scheme(ada::url_base&& u) noexcept;
-
-    /**
-     * @private
-     * Take the scheme from another URL. The scheme string is copied from the provided url.
-     */
-    inline void copy_scheme(const ada::url_base& u) noexcept;
-
-    /**
-     * @private
-     *
-     * Return a string representing the scheme. Note that get_scheme_type() should often be used instead.
-     * @see https://url.spec.whatwg.org/#dom-url-protocol
-     */
-    [[nodiscard]] inline std::string_view get_scheme() const noexcept;
-
-    /**
-     * Set the scheme for this URL. The provided scheme should be a valid
-     * scheme string, be lower-cased, not contain spaces or tabs. It should
-     * have no spurious trailing or leading content.
-     */
-    inline void set_scheme(std::string&& new_scheme) noexcept;
```

Comment:
I don't see practical improvement of removing setters and getters from base class as virtual functions.

Reply:
Because we want to have the freedom to implement them in the most efficient way possible... I tried to explain it in my comment to this PR:

> The problem is that ada::url_aggregator and ada::url have vastly different underlying implementation. Forcing both to conform to the same interface, increases the complexity. It also means that changes on one side (e.g., ada::url_aggregator) impact to the other side (ada::url). So tuning one piece of our code might later require that we update both side. It also might introduce artificial limits. For example, it is natural for ada::url to return an std::string when we ask for the href. However, for ada::url_aggregator, it is maybe more natural to just return a const std::string reference. Similarly, when asking for the url_components, it is natural to ask for a new instance when query ada::url but more natural to get a const reference when querying ada::url_aggregator.

Forcing both classes to follow the same interface is going to prevent you from optimizing both of the at their full potential.

It is easier, once both of them have been fully debugged and optimized to the refactor. I feel that locking up the interface for ada::url_aggregator on what ada::url has makes it difficult for us to both make progress (because everything is co-dependent) and I fear that the net result will have suboptimal performance.

I recommend we wait until we have high efficient implementations before deciding on whether a common interface is good, and deciding its exact shape.